src/loader.c

```diff
@@ -233,13 +230,17 @@ bool parse_peer(void *base, const char *name, const char *connstr)
 
 	free(peer->host);
 	peer->host = host;
+	host = NULL;
 	peer->port = port;
 	peer->pool_size = pool_size;
 
+	Assert(host == NULL);
+
```

Comment:
These additions seem a bit unnecessary.
```suggestion
	peer->port = port;
	peer->pool_size = pool_size;

```